pkg/apis/operator/v1alpha1/types_garden.go

```diff
@@ -623,6 +623,16 @@ type DashboardGitHub struct {
 
 // DashboardOIDC contains configuration for the OIDC settings.
 type DashboardOIDC struct {
+	// ClientIDPublic is the public client ID.
+	// Falls back to the API server's OIDC configuration if not set here.
+	// +kubebuilder:validation:MinLength=1
+	// +optional
+	ClientID *string `json:"clientID,omitempty"`
```

Comment:
Nit: I think it should be one of the two versions below to be consistent:
```suggestion
	// ClientID is the public client ID.
	// Falls back to the API server's OIDC configuration if not set here.
	// +kubebuilder:validation:MinLength=1
	// +optional
	ClientID *string `json:"clientID,omitempty"`
```

or

```suggestion
	// ClientIDPublic is the public client ID.
	// Falls back to the API server's OIDC configuration if not set here.
	// +kubebuilder:validation:MinLength=1
	// +optional
	ClientIDPublic *string `json:"clientIDPublic,omitempty"`
```

☝️ this version would be consistent with the dashboard's [`OIDCValues`](https://github.com/gardener/gardener/blob/bad9f34c55be9ddff98e5b4abfe8d2f71ef74cea/pkg/component/gardener/dashboard/dashboard.go#L80).

Reply:
I prefer `ClientID` which is also the value used in the Dashboard repo ([ref](https://github.com/gardener/dashboard/blob/c79afc80b9e6e189e1ce77c9b79011ce4db77f9e/charts/gardener-dashboard/values.yaml#L123)).

Reply:
> I prefer ClientID which is also the value used in the Dashboard repo ([ref](https://github.com/gardener/dashboard/blob/c79afc80b9e6e189e1ce77c9b79011ce4db77f9e/charts/gardener-dashboard/values.yaml#L123)).

This is the dashboard's clientID, not the public client ID, which would be this one here: [oidc.public.clientId](https://github.com/gardener/dashboard/blob/c79afc80b9e6e189e1ce77c9b79011ce4db77f9e/charts/gardener-dashboard/values.yaml#L142-L144)
These are the oidc helm chart values of the dashboard (some fields omitted)
```yaml
oidc:
  # well-known URL for OpenID Provider Issuer Discovery
  issuerUrl: ~
  # clientId is the public identifier of the gardener-dashboard aplication
  clientId: dashboard
  # clientSecret is the private secret of the gardener-dashboard aplication
  clientSecret: ~
  # configuration for kubeconfig download required by kubelogin
  public:
    # clientId is the identifier of the public oidc client use by kubelogin
    clientId: kube-kubectl
    # clientSecret is the public client secret use by kubelogin and all users
    clientSecret: ~
    # force PKCE usage (automatically enabled if no clientSecret is given)
    usePKCE: true
```

So I would also vote for `clientIDPublic`, otherwise it is confusing which clientId is used for which purpose.

Reply:
Thanks for clarifying @petersutter. I will adapt the PR.